autonomous.py

```diff
@@ -2,6 +2,24 @@
 
 import wpilib
 
+def drive_and_rotate(drivetrain, gyro):
+    yield from ArcadeAutonomous(drivetrain, 0.3, 0, 3).run()
+    yield from RotateAutonomous(drivetrain, gyro, 90, 0.5).run()
+    yield from ArcadeAutonomous(drivetrain, 0.3, 0, 3).run()
+
+class BaseAutonomous:
+    def init(self):
+        return self
+
+    def execute(self):
+        pass
+
+    def end(self):
+        pass
+
+    def run(self):
+        self.init()
```

Comment:
we should put in a nested function that does everything in execute then ends, like
```python
def _execute():
    yield from self.execute()
    self.end()
```
then we can `return _execute()` instead